### PR TITLE
Fix wrong returning state when there's no hid device present

### PR DESCRIPTION
**Pull request by BitBander**
hid device return error while it initialized failed

### CODE REVIEW

File: .settings/language.settings.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (luxianzi) [inline .settings/language.settings.xml:1]: What is this file used for? Is it for saving cross compiler settings when using PPSDK?

File: hal/source/hid_device.cpp [hunk 2]
```diff
@@ -61,6 +61,8 @@ error_condition HIDDevice::ReConfig(const CommonKeyPairs &config) {
 		SetInitialized();
 		break;
 	}
+	if (!IsInitialized())
```
Comment (luxianzi) [inline hal/source/hid_device.cpp:64]: Nice!

File: .gitignore [hunk 3]
```diff
@@ -2,3 +2,4 @@
 .cproject
 /build
 
+/Debug/
```
Comment (luxianzi) [inline .gitignore:5]: Also I tend to update and not to use current PPSDK in the future, if we are going to ignore `Debug` directory, could we also ignore `Release` which is also generated by Eclipse?

**Review by luxianzi — COMMENTED** [review]
Please also update the commit message to include more details of your fix. Thank you.

File: .settings/language.settings.xml (hunk 1 above)
Comment (BitBander) [reply .settings/language.settings.xml:1]: yes, it is

File: .gitignore (hunk 3 above)
Comment (BitBander) [reply .gitignore:5]: Yes, I'll add it to ignore list later

**Review by BitBander — COMMENTED** [review]
All changes are confirmed

**Review by BitBander — COMMENTED** [review]
approve

**Comment by BitBander** [comment]
> approve

**Comment by BitBander** [comment]
Approved

**Review by luxianzi — CHANGES_REQUESTED** [review]
Please fix the title of the commit.

**Comment by BitBander** [comment]
> Please fix the title of the commit.

OK

**Comment by BitBander** [comment]
> Please fix the title of the commit.

Title has been changed

File: .gitignore [hunk 4]
```diff
@@ -1,4 +1,6 @@
 .project
 .cproject
 /build
-
+/release
+/Debug/
+/Release/
```
Comment (luxianzi) [inline .gitignore]: Please always add a new line to the end of file.

**Review by luxianzi — APPROVED** [review]
LGTM

**Pull request merged**